### PR TITLE
docs: update migration data procedures

### PR DESCRIPTION
**Pull request by zxkane**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: docs/en/upgrade.md [hunk 1]
```diff
@@ -91,17 +91,36 @@ The solution automatically and asynchronously upgrades the views and materialize
...
+
+    ```sql
+    -- please replace <app-id> with your actual app id
+    -- update the day range(180 days in below example) based on your needs
+    CALL "<app-id>".clickstream_event_base_view_sp(NULL, NULL, 24*180);
+    ```
+
+    !!! info "Note"
+
+        If the volume of migrated data is large, it is recommended to refresh the `clickstream_event_base_view` in batches. That is, based on the occurrence time of the events, call the following stored procedure multiple times to refresh the data in batches.
```
Comment (luorobin-a2z) [inline docs/en/upgrade.md:112]: 最好能给一下多大 是large

**Review by luorobin-a2z — APPROVED** [review]

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.40|< 80|
|Duplicated lines density|:white_check_mark: OK|5.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|2|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1386)
###### _updated: 5/26/2024, 08:38:37 (UTC+0)_

**Comment by zxkane** [comment]
# SonarQube Code Analytics
## Quality Gate passed

passed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-1386&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-1386&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-1386&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

**Review by llmin — DISMISSED** [review]

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.50|< 80|
|Duplicated lines density|:white_check_mark: OK|5.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|2|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1386)
###### _updated: 5/27/2024, 01:55:35 (UTC+0)_

File: docs/en/upgrade.md [hunk 2]
```diff
@@ -91,17 +91,36 @@ The solution automatically and asynchronously upgrades the views and materialize
...
+
+    ```sql
+    -- please replace <app-id> with your actual app id
+    -- update the day range(180 days in below example) based on your needs
+    CALL "<app-id>".clickstream_event_base_view_sp(NULL, NULL, 24*180);
+    ```
+
+    !!! info "Note"
+
+        If new events come before completing this step or the volume of migrated data is large, it is recommended to refresh the `clickstream_event_base_view` in batches. That is, based on the occurrence time of the events, call the following stored procedure multiple times to refresh the data in batches(needs to be in order from old to new event dates).
```
Comment (luorobin-a2z) [inline docs/en/upgrade.md:112]: It is recommended to refresh the clickstream_event_base_view in batches, especially in the following scenarios:
- When there are new event load jobs coming in before the migration job completes.
- When the volume of migrated data is large (e.g., 100 millions in a batch).

Noe that refreshing the data in batches needs to be done based on the event timestamp. Call the following stored procedure multiple times, in order from old to new event timestamps.

**Review by luorobin-a2z — CHANGES_REQUESTED** [review]
refine the step 7 wording

**Review by luorobin-a2z — APPROVED** [review]

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.40|< 80|
|Duplicated lines density|:white_check_mark: OK|5.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|2|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1386)
###### _updated: 5/27/2024, 03:35:12 (UTC+0)_

**Pull request merged**